### ISSUE TITLE
Serializing beans without setter for a private field causes java.lang.IllegalAccessError: tried to access field

### ISSUE THREAD

**Issue by billybong** [netplex/json-smart-v2#49]
It seems like the check `Accessor::isPublic` is wrong since it checks if a field in a bean has a corresponding setter method. If not, the BeanAccessBuilder treats the field as public which causes IllegalAccessError when **serializing** the object.

It feels really weird that a setter should be required for serializing and it is a very common pattern to have immutable objects with only private fields and getters. The builder should have directed the accessor to use the getter instead in this case.

Gist with repro below:
https://gist.github.com/billybong/b0732928f5f891d67b066dc315c9b7ec

**Comment by Alanscut** [comment]
I found that the problem is caused by the `JSONObject.toString()` method:
```java
...
JSONObject object = new JSONObject(map); // is ok
System.out.println(object);  // will report an error, because it called toString() method
```
`JSONObject.toString()` is the process of deserialization, serialization requires a getter method in the POJO, so maybe its not an issue of json-smart.

**Comment by UrielCh** [comment]
that may be a bug ....

**Comment by UrielCh** [comment]
fixed for V 2.4.2

It should be improved to improve error messages.

**Comment by UrielCh** [comment]
2.4.2 is out.
